napari/layers/points/_tests/test_points.py

```diff
@@ -443,13 +444,19 @@ def test_remove_selected_updates_value():
     data = 20 * np.random.random(shape)
     layer = Points(data)
 
+    layer.events.data = Mock()
     # set the value
     layer._value = 3
     layer._value_stored = 3
 
-    layer.selected_data = {0, 5, 6, 7}
+    selection = {0, 5, 6, 7}
```

Comment:
If the frozen set cannot be used because of lost of selection then this line contains bug.

Reply:
I will write a separate test for this case. Thanks for spotting.

Reply:
Yeah here and there within functions it gets casted to a list to allow for sorting before deleting for example because otherwise it would mess with the sequence of deleting causing incorrect deletion

Reply:
ah here it actually is not a bug I believe. Reason for this is that internally the set is cast to a list and sorted in order correctly delete points. Same is done for shapes